runtime/builtin_types_test.go

```diff
@@ -422,3 +423,63 @@ func TestBuiltinSetAttr(t *testing.T) {
 		}
 	}
 }
+
+func TestRawInput(t *testing.T) {
+	fun := newBuiltinFunction("TestRawInput", func(f *Frame, args Args, _ KWArgs) (*Object, *BaseException) {
+		// Create a fake Stdin for input test.
+		stdinFile, _ := ioutil.TempFile(os.TempDir(), "prefix")
+		stdinFile.Write([]byte("hello grumpy\n"))
+		stdinFile.Close()
+		stdinFile, _ = os.Open(stdinFile.Name())
+		testStdin := NewFileFromFD(stdinFile.Fd())
+		oldStdin := Stdin
+		Stdin = testStdin
+
+		// Create a fake Stdout for output test.
+		r, w, err := os.Pipe()
+		if err != nil {
+			return nil, f.RaiseType(RuntimeErrorType, fmt.Sprintf("failed to open pipe: %v", err))
+		}
+
+		oldStdout := os.Stdout
+		os.Stdout = w
+		defer func() {
+			os.Stdout = oldStdout
+			Stdin = oldStdin
+			stdinFile.Close()
+			os.Remove(stdinFile.Name())
```

Comment:
Move `defer os.Remove(stdinFile.Name())` up to where you create the file so that an early return won't leave the tempfile lying around.